build/tf-wrapper.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+branch=$2
+policyrepo=$3
+
+prod_path="prod"
+shared_path="shared"
+
+tf_validator_path="./terraform-validator"
+
+##functions
+tf_apply() {
+  echo "*************** TERRAFORM APPLY *******************"
+  echo "      At environment: ${branchname} "
+  echo "***************************************************"
+  if [ -d "./envs/${branchname}" ]; then
+    cd "./envs/${branchname}" || exit
+    terraform apply -input=false -auto-approve "${branchname}.tfplan" || exit 1
+    cd ../.. || exit
+  else
+    echo "ERROR:  ${branchname} does not exist"
+  fi
+}
+
+tf_init() {
+  echo "*************** TERRAFORM INIT *******************"
+  echo "      At environment: ${branchname} "
+  echo "**************************************************"
+  if [ -d "./envs/${branchname}" ]; then
+    cd "./envs/${branchname}" || exit
+    terraform init || exit 11
+    cd ../.. || exit
+  else
+    echo "ERROR:  ${branchname} does not exist"
+  fi
+}
+
+tf_plan() {
+  echo "*************** TERRAFORM PLAN *******************"
+  echo "      At environment: ${branchname} "
+  echo "**************************************************"
+  if [ -d "./envs/${branchname}" ]; then
+    cd "./envs/${branchname}" || exit
+    terraform plan -input=false -out "${branchname}.tfplan" || exit 21
+    cd ../.. || exit
+  else
+    echo "ERROR:  ${branchname} does not exist"
+  fi
+}
+
+tf_validate() {
+  echo "*************** TERRAFORM VALIDATE ******************"
+  echo "      At environment: ${branchname} "
+  echo "      Using policy from: ${policyrepo} "
+  echo "*****************************************************"
+  if ! ${tf_validator_path} version &> /dev/null; then
+    echo "terraform-validator not found!  Check path or visit"
+    echo "https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#how-to-use-terraform-validator"
+  else
+    if [ -d "./envs/${branchname}" ]; then
+      cd "./envs/${branchname}" || exit
+      terraform show -json "${branchname}.tfplan" > "${branchname}.json" || exit 32
+      terraform-validator validate "${branchname}.json" --policy-path="${policyrepo}" || exit 33
```

Comment:
Follow up from https://github.com/terraform-google-modules/terraform-example-foundation/pull/91#pullrequestreview-451037450
As we are not doing a `git clone $policyrepo` Should we delegate the git clone of `policyrepo` to the CI systems and instead just expect the `policypath` here?

Reply:
@bharathkkb my bad, I forgot about your comment on this in the last attempt.  If there's a preference (clone it here or clone it in CI), let me know and I can modify as needed.

Reply:
I will defer to @rjerrems @morgante on what might be best practice. From CB perspective I think it might best to have everything packaged into one image, otherwise we will need to use another image for just git cloning the policy repo.

Reply:
I am don't have a lot of experience with terraform-validator, but it might be useful to clone the policy repo in the build - which customers can then swap out with there own custom policy repo, once they understand how it works. I think @morgante will have the best steer here

Reply:
I'd lean towards just having a policypath (environment) variable in this script and assuming the clone happens elsewhere. Git clone requires a different set of credentials which this image might not have.

Reply:
tracking in #120